gprofiler/main.py

```diff
@@ -267,6 +268,7 @@ def stop(self) -> None:
             prof.stop()
 
     def _snapshot(self) -> None:
+        self._container_names_client.refresh_container_names_cache()
```

Comment:
I think that we need to refresh container cache at the beginning of profiling. If profiled process will be terminated during profiling we won't be able to get his container name because there might be a situation when it will happen before first call of `get_container_name()` and there won't be any container_pid's in cache. I added a test to test_sanity.py to simulate such case.

Reply:
I see. Something is still off here for me, I will re-review thoroughly after you fix the other points I commented now

Reply:
This is not needed, I deleted this and check for every process at the beginning of perf snapshot